module/data/fields.mjs

```diff
@@ -150,6 +150,86 @@ export class IdentifierField extends foundry.data.fields.StringField {
 
 /* -------------------------------------------- */
 
+/**
+ * @typedef {StringFieldOptions} LocalDocumentFieldOptions
+ * @property {boolean} [fallback=false]  Display the string value if no matching item is found.
+ */
+
+/**
+ * A mirror of ForeignDocumentField that references a Document embedded within this Document.
+ *
+ * @param {typeof Document} model              The local DataModel class definition which this field should link to.
+ * @param {LocalDocumentFieldOptions} options  Options which configure the behavior of the field.
+ */
+export class LocalDocumentField extends foundry.data.fields.DocumentIdField {
+  constructor(model, options={}) {
+    if ( !foundry.utils.isSubclass(model, foundry.abstract.DataModel) ) {
+      throw new Error("A ForeignDocumentField must specify a DataModel subclass as its type");
+    }
+
+    super(options);
+    this.model = model;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * A reference to the model class which is stored in this field.
+   * @type {typeof Document}
+   */
+  model;
+
+  /* -------------------------------------------- */
+
+  static get _defaults() {
+    return foundry.utils.mergeObject(super._defaults, {
+      nullable: true,
+      readonly: false,
+      idOnly: false,
+      fallback: false
+    });
+  }
+
+  /* -------------------------------------------- */
+
+  _cast(value) {
+    if ( typeof value === "string" ) return value;
+    if ( (value instanceof this.model) ) return value._id;
+    throw new Error(`The value provided to a LocalDocumentField must be a ${this.model.name} instance.`);
+  }
+
+  /* -------------------------------------------- */
+
+  _validateType(value) {
+    if ( !this.options.fallback ) super._validateType(value);
+  }
+
+  /* -------------------------------------------- */
+
+  initialize(value, model, options={}) {
+    if ( this.idOnly ) return value;
```

Comment:
I think this somewhat violates the semantics of the `DocumentIdField`. If this is a field with `fallback: false`, then the value returned in the `idOnly` case might be some arbitrary string and not a document ID. We can't check to see if the document exists first, since `idOnly` can be expected to return an ID even if the document doesn't exist. I think we might have to check `isValidId` here.
```suggestion
    if ( this.idOnly ) return this.options.fallback || foundry.data.validators.isValidId(value) ? value : null;
```